pkg/sql/planner.go

```diff
@@ -93,7 +93,7 @@ type extendedEvalContext struct {
 	Jobs *jobsCollection
 
 	// SchemaChangeJobCache refers to schemaChangeJobsCache in extraTxnState.
-	SchemaChangeJobCache map[descpb.ID]*jobs.Job
+	SchemaChangeJobCache map[descpb.ID]*jobs.Specification
```

Comment:
My guess is that we're going to want to nuance this data structure a bit. There are places where we delete from this cache. Those places are trying to be a bit clever and used to rely on the fact that we had already written the job. I'm thinking what we want is to create a slice of job records and a map from descriptor ID to the record. Then in the places where today we delete from the map we'd retain the record in the slice. 

I suspect this is the cause of some of the failing tests.